packages/components/psammead-media-player/src/Canonical/useTimeout.jsx

```diff
@@ -0,0 +1,26 @@
+import { useRef, useEffect } from 'react';
+
+/**
+ * useTimeout Hook.
+ * @param {function} callback A callback function that returns `true` or `false`.
+ * @param {object} iframeRef A React DOM reference to an iframe.
+ * @param {number} timeout The number of milliseconds until the timeout returns `true`.
+ */
+const useTimeout = (callback, iframeRef, timeout) => {
+  const timer = useRef(null);
+
+  const handleLoad = () => {
+    clearTimeout(timer.current);
+    callback(false);
+  };
+
+  useEffect(() => {
+    iframeRef.current.addEventListener('load', handleLoad, { once: true });
+    timer.current = setTimeout(() => {
+      callback(true);
+    }, timeout);
+    return () => clearTimeout(timer.current);
```

Comment:
Although modern browsers will handle this, I think we should also be removing the event listener in this callback during unmounting. 
 ```suggestion
    return () => { 
       clearTimeout(timer.current);
       iframeRef.current.removeEventListener('load', handleLoad);
     }
```

Reply:
Have you tried this? `iframeRef` will be `null` if this function gets called, because the component will have already been unmounted. I thought about storing a further reference to it, but I feel this is unnecessary, because:
- The garbage collector will clean this up for us.
- It is mitigated by the fact addEventListener is called with `once: true`.